### ISSUE TITLE
Strange Compilation Error When Referencing 3.1.0.1

### ISSUE THREAD

**Issue by thefringeninja** [TalAloni/StandardSocketsHttpHandler#9]
Continuing from #8 

I get the following error when adding this as a reference to my project and compiling:
```
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetStatus' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.get_Version' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.OnCompleted' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.Reset' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.SetException' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs(8,76): error CS0656: Missing compiler required member 'System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.SetResult' [/home/jwow/dev/EventStore-Client-Dotnet/src/EventStore.Client/EventStore.Client.csproj]
```

Source code of [AsyncStreamReaderExtensions.cs](https://github.com/EventStore/EventStore-Client-Dotnet/blob/master/src/EventStore.Client.Common/AsyncStreamReaderExtensions.cs)

**Comment by TalAloni** [comment]
Hi,
I can't take responsibility for your project and your codebase.
I was able to utilize the library from .NET Framework 4.7.2 and from .NET Core 3.1
What is your target? .NET Standard 2.0? 2.1?

**Comment by thefringeninja** [comment]
The target is `net48`.

**Comment by TalAloni** [comment]
Do you have any issue with a .NET 4.8 Console application? or just with your project?

**Comment by thefringeninja** [comment]
I've created a minimal repro which illustrates the problem: https://github.com/thefringeninja/standardsocketshandler-repro. It fails to compile if it uses `IAsyncEnumerable`.

I'm starting to think now that it's because this library and Microsoft.Bcl.AsyncInterfaces both define [ManualResetValueTaskSourceCore](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Bcl.AsyncInterfaces/src/System/Threading/Tasks/Sources/ManualResetValueTaskSourceCore.cs).

**Comment by TalAloni** [comment]
Your change is implemented in 3.1.0.2
https://www.nuget.org/packages/StandardSocketsHttpHandler/3.1.0.2

**Comment by thefringeninja** [comment]
Thanks!
